src/material/button/button-lazy-loader.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DOCUMENT} from '@angular/common';
+import {ANIMATION_MODULE_TYPE, Injectable, NgZone, OnDestroy, inject} from '@angular/core';
+import {
+  MAT_RIPPLE_GLOBAL_OPTIONS,
+  RippleConfig,
+  RippleGlobalOptions,
+  RippleRenderer,
+  RippleTarget,
+} from '@angular/material/core';
+import {Platform} from '@angular/cdk/platform';
+
+/** The options for the MatButtonRippleLoader's event listeners. */
+const OPTIONS = {capture: true};
+
+/** The attribute attached to a mat-button whose ripple has not yet been initialized. */
+const MAT_BUTTON_RIPPLE_UNINITIALIZED = 'mat-button-ripple-uninitialized';
+
+@Injectable({providedIn: 'root'})
+export class MatButtonLazyLoader implements OnDestroy {
+  private _document: Document | null = inject(DOCUMENT, {optional: true});
+  private _animationMode: string | null = inject(ANIMATION_MODULE_TYPE, {optional: true});
+  private _globalRippleOptions: RippleGlobalOptions | null = inject(MAT_RIPPLE_GLOBAL_OPTIONS, {
+    optional: true,
+  });
```

Comment:
```suggestion
  private _document = inject(DOCUMENT, {optional: true});
  private _animationMode = inject(ANIMATION_MODULE_TYPE, {optional: true});
  private _globalRippleOptions = inject(MAT_RIPPLE_GLOBAL_OPTIONS, {
    optional: true,
  });
```

I think you can omit these explicit types and the types will be inferred